### PR TITLE
feat: add keep alive support

### PR DESCRIPTION
**Pull request by khshah6**
Resolves: #950 

**Description**
This fix enables consumers to reuse connections by supporting [keep-alive](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Connection#Syntax).

When `keepAlive` is set to true, SendGrid will use [forever-agent](https://github.com/request/forever-agent) (HTTP Agent that keeps socket connections alive between keep-alive requests).

This has been tested in a simple node request. End users do not have to provide any additional information when calling SendGrid for a client or to send mail and as such has no breaking change.

Linked issue includes examples of similar changes made in Stripe and Twilio apis.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/sendgrid/sendgrid-nodejs/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://support.sendgrid.com), or create a GitHub Issue in this repository.

### CODE REVIEW

File: packages/mail/src/classes/mail-service.js [hunk 1]
```diff
@@ -40,6 +41,17 @@ class MailService {
     return this;
   }
 
+  /**
+   * Reuse connections
+   */
+  setKeepAlive(keepAlive) {
+    if (keepAlive === false) {
+      return;
```
Comment (eshanholtz) [inline packages/mail/src/classes/mail-service.js:49]: What about turning off keep alive? If this just returns, you can never "unset" it.

File: packages/mail/src/classes/mail-service.js [hunk 2]
```diff
@@ -40,6 +41,17 @@ class MailService {
...
+    }
+
+    this.client.setDefaultRequest('agent', new Agent({ keepAlive: true }));
```
Comment (eshanholtz) [inline packages/mail/src/classes/mail-service.js:52]: I think the key needs to be `httpsAgent`, according to the [documentation](https://github.com/axios/axios#request-config). This logic should be moved to the underlying `Client` with a separate helper method that we can call from here, similar to calling [`this.client.setApiKey`](https://github.com/khshah6/sendgrid-nodejs/blob/khshah6-add_keep_alive_support/packages/client/src/classes/client.js#L38) from [`setApiKey`](https://github.com/sendgrid/sendgrid-nodejs/pull/1226/files#diff-a4d6dd09858240682deb6dc3cf67bdda6a4b3b5e041537940d56a5f42b7c9a51R38).

Comment (khshah6) [reply packages/mail/src/classes/mail-service.js:52]: sure, good call out

File: packages/mail/src/classes/mail-service.js (hunk 1 above)
Comment (khshah6) [reply packages/mail/src/classes/mail-service.js:49]: `keepAlive` is already false by default, its an opt-in feature

**Comment by JenniferMah** [comment]
Closing until PR feedback is addressed.

**Pull request closed without merging**